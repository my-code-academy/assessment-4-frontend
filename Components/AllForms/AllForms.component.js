import React, { Component } from 'react';
import {
  TextInput, TouchableOpacity, Text, KeyboardAvoidingView, Image, ScrollView, StyleSheet, View,
} from 'react-native';
import axios from 'axios';
import FormCard from '../FormCards/FormCard.component';

export default class AllForms extends Component {
  state = {
    allFormsData: [],
  }

  componentDidMount() {
    const allForms = axios.get('http://localhost:7777/forms/names').then((formsNames) => {
      this.setState({ allFormsData: formsNames.data });
    });
  }

  render() {
    const { allFormsData } = this.state;
    return (
      <ScrollView contentContainerStyle={styles.allCards}>
        <View>
          {
            allFormsData.map(formData => (
              <TouchableOpacity>
                <FormCard formName={formData.name} datetime={formData.date} />
              </TouchableOpacity>
            ))
          }
        </View>

      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  allCards: {
    // flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'center',
    backgroundColor: '#D8E4F0',
    // height: 800,
  },
});
