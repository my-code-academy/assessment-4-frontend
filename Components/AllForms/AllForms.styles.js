import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  allCards: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'center',
    backgroundColor: '#D8E4F0',
    // height: 800,
  },
  MainContainer: {
    backgroundColor: '#D8E4F0',
  },
  TouchableOpacityStyle: {
    width: 50,
    height: 50,
    alignItems: 'center',
    justifyContent: 'center',
    // right: 30,
    // bottom: 30,
  },
  FloatingButtonStyle: {
    resizeMode: 'contain',
    width: 50,
    height: 50,
  },
});

export default styles;
