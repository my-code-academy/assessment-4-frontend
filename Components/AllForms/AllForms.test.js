import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AllForms from './AllForms.component';

Enzyme.configure({ adapter: new Adapter() });

describe('AllForms', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<AllForms />);
    expect(tree).toMatchSnapshot();
  });

  it('Should call componentDidMount once', async (done) => {
    const wrapper = shallow(<AllForms />);
    expect(wrapper.instance().state.allFormsData).toEqual({});
    await wrapper.instance().componentDidMount();
    expect(wrapper.instance().state.allFormsData.length).not.toEqual(0);
    done();
  });
});

