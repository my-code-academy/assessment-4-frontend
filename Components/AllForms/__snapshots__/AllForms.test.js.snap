// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllForms should match the snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "backgroundColor": "#D8E4F0",
      "justifyContent": "flex-start",
    }
  }
>
  <View>
    <View />
  </View>
</RCTScrollView>
`;
